tests/flight/sync.rs

```diff
@@ -126,40 +149,47 @@ async fn test_sync_happy_path() -> std::result::Result<(), Box<dyn std::error::E
     let results = ctx.collect(plan.clone()).await?;
 
     let expected = [
-        "+-------+-------+-------+---------------------+",
-        "| col_1 | col_2 | col_3 | col_4               |",
-        "+-------+-------+-------+---------------------+",
-        "| 1     | one   | 1.0   | 2001-01-01T01:01:01 |",
-        "| 2     | two   | 2.0   | 2002-02-02T02:02:02 |",
-        "| 3     |       | 3.0   |                     |",
-        "| 4     |       | 4.0   |                     |",
-        "+-------+-------+-------+---------------------+",
+        "+----+-------+-------+---------------------+-------+",
+        "| c1 | c2    | c3    | c4                  | c5    |",
+        "+----+-------+-------+---------------------+-------+",
+        "| 1  | one   | 1.111 | 2001-01-01T01:01:01 |       |",
+        "| 2  | two   | 2.0   | 2002-02-02T02:02:02 | false |",
+        "| 3  | three | 3.0   |                     | false |",
+        "| 4  | four  | 4.0   |                     | false |",
+        "+----+-------+-------+---------------------+-------+",
     ];
 
-    assert_batches_eq!(expected, &results);
+    assert_batches_sorted_eq!(expected, &results);
 
+    //
     // Sync #3; this will be held in memory
+    //
+
     let schema = Arc::new(Schema::new(vec![
-        Field::new(
-            "col_4",
-            DataType::Timestamp(TimeUnit::Microsecond, None),
-            true,
-        ),
-        Field::new("col_2", DataType::Utf8, true),
-        Field::new("col_1", DataType::Int32, true),
+        Field::new("c4", DataType::Timestamp(TimeUnit::Microsecond, None), true),
+        Field::new("c2", DataType::Utf8, false),
+        Field::new("c1", DataType::Int32, false),
         Field::new(SEAFOWL_SYNC_DATA_UD_FLAG, DataType::Boolean, true),
     ]));
 
+    // Have one row (5) be an append, followed by delete followed by append
     let batch = RecordBatch::try_new(
         schema.clone(),
         vec![
             Arc::new(TimestampMicrosecondArray::from(vec![
-                1115269505000000,
-                1149573966000000,
+                None,
+                None,
+                Some(1115269505000000),
+                Some(1149573966000000),
+            ])),
+            Arc::new(StringArray::from(vec![
+                Some("five"),
+                Some("five"),
+                Some("five"),
```

Comment:
The second one should be None (since we're deleting it) and I suggest the final one should be different so that we can be confident that the last value won? Or does the fact that the timestamp goes from None -> 1115269505000000 test this?

```suggestion
                None,
                Some("five_updated_after_delete"),
```

Reply:
> The second one should be None (since we're deleting it)

Can't do that now since I've made c2 be a part of PK as well.

> Or does the fact that the timestamp goes from None -> 1115269505000000 test this?

Exactly, only in the last row do we get an actual timestamp (which is shown in the query results).